### PR TITLE
Deprecate ObjectWrapper API

### PR DESCRIPTION
**Pull request by TimPansino**
# Overview
## Context 

`ObjectWrapper` was previously used in instrumentation but was simply a custom and more manual version of `FunctionWrapper` from `wrapt`. There is no functional difference in how it is used today in the agent and should be deprecated and removed.

## Changes

* Deprecate `ObjectWrapper` API.
* Add `CallableObjectProxy` to public APIs to replace some of the functionality of `ObjectWrapper`
* Add missing `NewRelicContextFormatter` supportability wrapping in `newrelic.agent`.
* Remove instrumentation references to `ObjectWrapper`, replacing with `FunctionWrapper` or `ObjectProxy`.
* Remove custom `ObjectWrapper` implementation and instead create a pass through to `FunctionWrapper`.

### CODE REVIEW

**Review by hmstepanek — CHANGES_REQUESTED** [review]
In general this looks good. I had one question about the import path change. It looks correct to me but I think we should change all of them or none of them as there are a lot more occurrences than just this one.

File: newrelic/hooks/framework_webpy.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 import newrelic.api.in_function
 import newrelic.api.out_function
 import newrelic.api.pre_function
-from newrelic.api.object_wrapper import callable_name
+from newrelic.common.object_wrapper import callable_name
```
[the remark this answers is not in the document]
Comment (TimPansino) [reply newrelic/hooks/framework_webpy.py:24]: I've changed all these references now to point to the new object wrapper location

File: newrelic/hooks/framework_pylons.py [hunk 2]
```diff
@@ -16,14 +16,14 @@
 import newrelic.api.transaction_name
 import newrelic.api.function_trace
 import newrelic.api.error_trace
-import newrelic.api.object_wrapper
+import newrelic.common.object_wrapper
```
Comment (hmstepanek) [inline newrelic/hooks/framework_pylons.py:19]: All of these callable name usages should import from `newrelic.common.object_names`? At this point, callable_name isn't even imported in object_wrapper. The pylint linter caught this issue if you look at the megalinter logs. The tests didn't fail though cause we don't actually have tests for this file.

**Review by hmstepanek — CHANGES_REQUESTED** [review]
It looks like we are importing callable_name from object_wrappers when it should be importing from object_names. Note the linter caught this issue but the tests did not. One of those rare cases where the linter caught something super important.

File: newrelic/hooks/framework_pylons.py (hunk 2 above)
Comment (hmstepanek) [reply newrelic/hooks/framework_pylons.py:19]: I just pushed up a couple fixes for this.

**Review by hmstepanek — APPROVED** [review]
I think this looks good now. Maybe just have a quick review of the commits I made before merging as a double check?

**Pull request merged**